### PR TITLE
[WIP] Add test for a percentage of rows to be null

### PR DESCRIPTION
**Pull request by danhphan**
This [Work in Progress] PR addresses the issue #238 that will allow for percentage of rows to be null.

CC @clausherther Please let me know if it needs any further changes.

Thank you.

### CODE REVIEW

**Comment by danhphan** [comment]
FYI: I have added a column named `col_null_percentage` in the `data_test.sql` file for testing purpose. This column have 4 rows including one null row. So the null percentage will be 1/4 = 0.25. And in the `integration_tests/models/schema_tests/schema.yml` file, added the test for that column will less than 0.3. Thank you.

```
      - name: col_null_percentage
        tests:
          - dbt_expectations.expect_column_null_percentage_to_be_less_than:
              value: 0.3
```

**Comment by danhphan** [comment]
Hi @clausherther I am not sure why the CircleCI is still running. I run all `dbt test` on a local machine and it was quite fast (a couple of minutes).

Any suggestions on this. Thank you!

**Comment by clausherther** [comment]
@danhphan I have to approve it first, will do now.

**Comment by clausherther** [comment]
I'll take a look at the PR itself this weekend and give feedback then.

**Comment by clausherther** [comment]
@danhphan I think you'll need to rebase to get CI to pass (I remove SQLFluff support for now since their latest release broke things). I wasn't able to rebase your branch from my side.

**Review by clausherther — CHANGES_REQUESTED** [review]

**Comment by clausherther** [comment]
I was able to rebase your branch, but left comments re: the test.
I think essentially, we'd need to add those tolerance params to `expression_is_true`, which is a more extensive change than maybe you had in mind. Let me know if you want to try and tackle that, or if you'd rather have me help with that.

**Comment by danhphan** [comment]
Hi @clausherther, yes, I think adding these tolerance params will allow better generalisation.

Let me work on that. If I face any issues, I will ask for your help :) Thank you.

File: macros/schema_tests/aggregate_functions/expect_column_null_percentage_to_be_less_than.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{% test expect_column_null_percentage_to_be_less_than(model,
```
[the remark this answers is not in the document]
Comment (danhphan) [reply macros/schema_tests/aggregate_functions/expect_column_null_percentage_to_be_less_than.sql:1]: Yes, agree!

**Comment by danhphan** [comment]
I am thinking to change the [expression_is_true](https://github.com/calogica/dbt-expectations/blob/bd7746d8c818d5060446e21a959a15f6aa26dc10/macros/schema_tests/_generalized/expression_is_true.sql#L1) test to:

```
with grouped_expression as (
    ...
),

tolerance_expression as (
    select ( sum (expression is false) * 1.0 / count(*) ) < {{ tolerance_percent }} as tolerated_expression
    from
        grouped_expression    
) ,

validation_errors as (
    select
        *
    from
        tolerance_expression
    where
        not(tolerated_expression {{ test_condition }})
)

```

But not sure if it will work for the case when we have a `group_by_columns` parameter.

In addition, is there a test in the codebase for the case of using `group_by_columns` parameter in the `expression_is_true` test?

Thank you!

**Comment by jenna-jordan** [comment]
Just curious why this PR got closed, and whether there are any plans to add this feature? (we commonly test that, for example, >90% of values are not null, using Great Expectations)
